validator/sawtooth_validator/networking/interconnect.py

```diff
@@ -758,6 +758,20 @@ def get_connection_status(self, connection_id):
             return connection_info.status
         return None
 
+    def is_connection_handshake_complete(self, connection_id):
+        """
+        Indicates whether or not a connection has completed the authorization
+        handshake.
+
+        Returns:
+            bool - True if the connection handshake is complete, False
+                otherwise
+        """
+        if connection_id in self._connections:
+            connection_info = self._connections[connection_id]
+            return connection_info.status == ConnectionStatus.CONNECTED
+        return False
```

Comment:
I think you need to acquire `self._connections_lock` here.

Reply:
Also, I think we prefer `try/except` to `if .. in ..`..

Reply:
None of those methods that look at that map use that.  Will put up a separate commit to refactor those to use a try/catch and the lock.